elements/lisk-cryptography/src/bls_lib/lib.ts

```diff
@@ -123,8 +145,14 @@ export const blsPopProve = (sk: Buffer): Buffer => {
 	const message = blsSkToPk(sk);
 	const sig = new blst.P2();
 
+	const secretKey = SecretKey.fromBytes(sk);
+
+	if (!isSecretKeyNonZeroModEC(secretKey)) {
+		throw new Error('Secret key is not valid.');
+	}
```

Comment:
If we remove this check from here, `SecretKey.fromBytes(sk)` on L148 is not checked for compliance that it should be non-zero modulo the group order.

Reply:
I'm not sure I understand. In `blsSkToPk` you do already
```js
        const secretKey = SecretKey.fromBytes(sk);

	if (!isSecretKeyNonZeroModOrder(secretKey)) {
		throw new Error('Secret key is not valid.');
	}
```
and `blsSkToPk` is called in line 143. So, I'm sure you could remove lines 148-150 and all tests for `blsPopProve` should pass including the test for`sk = groupOrder`.

BTW: It would be good to add the test for `blsPopProve` with `sk=0`.

Reply:
`sk` is only passed as an arg to `blsSkToPk` at L143. if we remove the lines 148-150, then the `sk` at L146 is not checked or it's safe to assume that after L143, we do not need to check if `SecretKey.fromBytes(sk);` is "safe"?